File: DSA/Vikash_kumar/Day_1_Question_1.py
def birthday_candy(candy, day, month):
    count = 0
    candy_sum = 0
    candy_size = 0
    
    for i in range(len(candy)):
        candy_sum += int(candy[i])
        candy_size += 1
        
        if candy_size > month:
            candy_sum -= int(candy[i - month])
            candy_size -= 1
        
        if candy_size == month and candy_sum == day:
            count += 1
    return count

candy = []
candy = input("Enter the number written on squares of candy with a space : ").split(" ")
day = int(input("Enter the Raju's birth day : "))
month = int(input("Enter the Raju's birth month : "))
print(birthday_candy(candy, day, month)) 

File: DSA/Rakshit Singh/Day 1/Day_1_Question_1.java
class Day_1_Question_1{
    public int max_number_of_ways(int[] candy,int day,int month){
        int len=candy.length;
        int ways=0;
        
        for(int x=0;x<len-month;x++)
        {
            int count=0;
            for(int y=0;y<month;y++){
                count+=candy[x+y];
            }
            if(count==day){
                ways++;
            }
        }
        return ways;
    }
}